### PR TITLE
Use serde_codegen instead of serde_macros

### PR DESCRIPTION
**Pull request by nox**

### CODE REVIEW

**Comment by nox** [comment]
I had to change `NativeFontHandle` to be `()` on non-OS X platforms, instead of a unit struct.

File: src/api.rs [hunk 1]
```diff
@@ -2,79 +2,21 @@
...
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+use {ApiMsg, AuxiliaryLists, BuiltDisplayList, ColorF, DisplayListId, Epoch};
+use {FontKey, IdNamespace, ImageFormat, ImageKey, NativeFontHandle, PipelineId};
+use {RenderApiSender, ResourceId, ScrollEventPhase, ScrollLayerState};
+use {StackingContext, StackingContextId, WebGLContextId, WebGLCommand};
 use byteorder::{LittleEndian, WriteBytesExt};
-use display_list::{AuxiliaryLists, AuxiliaryListsDescriptor, BuiltDisplayList};
-use display_list::{BuiltDisplayListDescriptor};
+use display_list::{};
```
Comment (pcwalton) [inline src/api.rs]: Remove this?

**Comment by nox** [comment]
@pcwalton r?

**Comment by nox** [comment]
I worked around the `NativeFontHandle` issue thanks to serde 0.7.11.

**Comment by nox** [comment]
Oops, this require an Euclid bump in offscreen_gl_context.

**Comment by nox** [comment]
https://github.com/emilio/rust-offscreen-rendering-context/pull/58

**Comment by nox** [comment]
The offscreen-rendering-context PR landed and @pcwalton told me r=me.

@bors-servo r=pcwalton

**Pull request merged**